runtime/doc/eval.txt

```diff
@@ -3507,9 +3507,22 @@ getcurpos()	Get the position of the cursor.  This is like getpos('.'), but
 			MoveTheCursorAround
 			call setpos('.', save_cursor)
 <
-							*getcwd()*
-getcwd()	The result is a String, which is the name of the current
-		working directory.
+getcwd([{window}[, {tab}]])					*getcwd()*
+		With no argument the result is a String, which is the name of the
+		current effective working directory. If an arguments are provided for a
+		window or a tabpage the working directory of that scope is returned.
+
+		Tabs and windows are identified using their respective number. If a
+		number is 0 the current tab or window is implied. If one is not given
+		it defaults to 0. Thus
+		
+			getcwd()
+			getcwd(0)
+			getcwd(0, 0)
+
+		are all equal. If {window} is -1 it will be ignored and only the tab
+		will be looked at.
```

Comment:
@HiPhish I understand the semantics of working directories (I think) and by extension of `getcwd()` – my question was about `haslocladir()`, which returns a Boolean value. To get back to my question:

> what does `haslocaldir(winnr, 0)` return when a tab local working directory is set for the tab containing `winnnr`, but not a window local one

– 1 (because there is a local working directory set for the current tab) or 0 (because there is no local working directory set for the window)?